### DISCUSSION TITLE
DAI 2024-2025 - Practical work 1 - Kilian Froidevaux and Nicolas Bovard

### DISCUSSION THREAD

**Discussion by nicolasbvd** [heig-vd-dai-course/heig-vd-dai-course#373] [Show and tell]
Our project is to manipulate bitmap image. We could create resolution change or filters such as blurring, 3D effect, black and white, zoom.

**Comment by ludelafo** [comment]
Looks good to me, thank you!

Do you agree @GeraudSilvestri?

**Reply by GeraudSilvestri** [reply]
All good for me 👍

**Reply by kilianfroideva** [reply]
@GeraudSilvestri @ludelafo 
Sorry for requesting a bit of your time but we have a question. Is it allowed to do the following for handeling bitmap files?  
`try (InputStream input = new FileInputStream(filePath)) {
            BufferedImage image = ImageIO.read(input);`

Because in the course, it says:
- You can only use the Java classes seen in the course to process the files (you can use other libraries to help you once the files are opened) 

And we are not sure if the above's code is respecting this constraint.

**Reply by GeraudSilvestri** [reply]
It is not, you are not allowed to use any external library to read or write the files. Once the files are read and loaded in your memory, you can use a library to modify it, but only from that point.

The goal of this work is to help you familiarize with the java.io library. The BufferedImage class you're trying to use is from the java.awt.Image class which isn't the same. In addition, from what me and @ludelafo have seen, the bufferedimage class is designed towards desktop apps to display an image, which is why the doc speaks of "rashers" and "renderers".

Hope this answered your question, if not, feel free to contact us again

**Comment by kilianfroideva** [comment]
Project description: Manipulate bitmap file to do grey-scaling filter, blur effect, zooming and resolution change.
Link: [release](https://github.com/kilianfroideva/heig-vd-dai-practical-content-1/tree/release)
Commit: [b56a14eb853e6ef09d2aa9e65b81aa172cd30e8a](https://github.com/kilianfroideva/heig-vd-dai-practical-content-1/commit/b56a14eb853e6ef09d2aa9e65b81aa172cd30e8a)
